components/function-controller/internal/controllers/serverless/function_reconcile.go

```diff
@@ -169,51 +169,73 @@ func (r *FunctionReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
 
 	gitOptions, err := r.readGITOptions(ctx, instance)
 	if err != nil {
-		return r.updateStatusWithoutRepository(ctx, ctrl.Result{}, instance, serverlessv1alpha1.Condition{
+		log.Error(err, "Reading git options failed")
+		if updateErr := r.updateStatusWithoutRepository(ctx, instance, serverlessv1alpha1.Condition{
 			Type:               serverlessv1alpha1.ConditionConfigurationReady,
 			Status:             corev1.ConditionFalse,
 			LastTransitionTime: metav1.Now(),
 			Reason:             serverlessv1alpha1.ConditionReasonSourceUpdateFailed,
 			Message:            fmt.Sprintf("Reading git options failed: %v", err),
-		})
+		}); updateErr != nil {
+			return ctrl.Result{}, errors.Wrap(updateErr, "while updating status")
+		}
+		return ctrl.Result{}, err
 	}
 
 	revision, err := r.syncRevision(instance, gitOptions)
 	if err != nil {
+		log.Error(err, "Syncing git revision failed")
 		result, errMsg := NextRequeue(err)
-		return r.updateStatusWithoutRepository(ctx, result, instance, serverlessv1alpha1.Condition{
+		if updateErr := r.updateStatusWithoutRepository(ctx, instance, serverlessv1alpha1.Condition{
 			Type:               serverlessv1alpha1.ConditionConfigurationReady,
 			Status:             corev1.ConditionFalse,
 			LastTransitionTime: metav1.Now(),
 			Reason:             serverlessv1alpha1.ConditionReasonSourceUpdateFailed,
 			Message:            errMsg,
-		})
+		}); updateErr != nil {
+			return ctrl.Result{}, errors.Wrap(updateErr, "while updating status")
+		}
+		return result, err
```

Comment:
This was originally cleaned up, but this block has fixes for two issues. To simplify review, I split the fixes in two PRs, this block is cleaned up in another PR.

Reply:
Please don't break the logic here.
You have created separate PR for this part of logic.